packages/nightingale-sequence-heatmap/src/nightingale-sequence-heatmap.ts

```diff
@@ -0,0 +1,451 @@
+    });
+    hm.setZooming({ axis: "x" });
+    hm.setVisualParams({ xGapPixels: 0, yGapPixels: 0 });
+    this.heatmapInstance = hm;
+
+    this.bindHeatmapEvents();
+
+    this.heatmapInstance.render(this.heatmapId!);
+    this.heatmapInstance.events.render.subscribe((_) => {
+      if (this["display-start"] !== 0 && this["display-end"] !== 0)
+        this.triggerHeatmapZoom();
+    });
+  }
+
+  /**
+   * Function to bind zoom and hover events between Heatmap component and Nightingale
+   */
+  bindHeatmapEvents() {
+    if (!this.heatmapInstance) return;
+    this.heatmapInstance.events.zoom.subscribe((d) => {
+      if (!d) return;
+      // On heatmap zoom dispatch event to Protvista
+      if (d.xMin + 0.5 !== this["display-start"]) {
+        this.dispatchEvent(
+          new CustomEvent("change", {
+            detail: {
+              value: d.xMin + 0.5,
+              type: "display-start",
+            },
+            bubbles: true,
+          }),
+        );
+      }
+      if (d.xMax - 0.5 !== this["display-end"]) {
+        this.dispatchEvent(
+          new CustomEvent("change", {
+            detail: {
+              value: d.xMax - 0.5,
+              type: "display-end",
+            },
+            bubbles: true,
+          }),
+        );
+      }
+    });
+
+    this.heatmapInstance.events.hover.subscribe((d) => {
+      if (!d) return;
+      // On heatmap zoom dispatch event to Protvista
+      this.dispatchEvent(
+        new CustomEvent("change", {
+          detail: {
+            value: `${d.xIndex + 1}:${d.xIndex + 1}`,
+            type: "highlight",
+          },
+          bubbles: true,
+        }),
+      );
+    });
+  }
+
+  /**
+   * Function to trigger Heatmap zooming from Nightingale
+   */
+  triggerHeatmapZoom() {
+    const toStart = this["display-start"]!;
+    const toEnd = this["display-end"]!;
+    if (this.heatmapInstance) {
+      this.heatmapInstance.zoom({
+        xMin: toStart - 0.5,
+        xMax: toEnd + 0.5,
+      });
+    }
+  }
+
+  /**
+   * Function to trigger Heatmap highlighting from Nightingale
+   */
+  triggerHeatmapHighlight() {
+    const highlight = this["highlight"];
+
+    if (this.heatmapInstance) {
+      // any so we can use private marker attributes
+      // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+      const heatmapInstanceMarker = <any>(
+        this.heatmapInstance.extensions.marker!
+      );
+
+      if (!highlight) {
+        // highlight null means no highlight to be shown
+        heatmapInstanceMarker.state.dom.svg
+          .selectAll("." + HeatmapClassNames.Marker)
+          .remove();
+        heatmapInstanceMarker.state.dom.svg
+          .selectAll("." + HeatmapClassNames.MarkerX)
+          .remove();
+        heatmapInstanceMarker.state.dom.svg
+          .selectAll("." + HeatmapClassNames.MarkerY)
+          .remove();
+        return;
+      }
+
+      // parse highlight start and end residues as indexes
```

Comment:
The highlight attribute could have multiple segments, for instance: `highlight="20:25,30:35"`. 
Have a look to `this.highlightedRegion.segments` where the multiple segments are already parsed. That should be available already here as you are implementing `withHighlight.

Reply:
Refactored so it's fixed, thanks.